src/mpi.jl

```diff
@@ -12,3 +12,17 @@ struct MPICommsContext{D <: AbstractDevice, C} <: AbstractCommsContext
 end
 
 function MPICommsContext end
+
+"""
+    local_communicator(ctx::MPICommsContext)
+
+Internal function to create a new MPI communicator for processes on the same physical node.
+
+The communicator must be freed by `MPI.free`
+```
+local_comm = ClimaComms.local_communicator(ctx)
+# use the communicator
+MPI.free(local_comm)
```

Comment:
Given this constraint, maybe we should support and use the `do` sytax to ensure that it is properly freed

Reply:
Agreed